src/dpp/guild.cpp

```diff
@@ -112,6 +112,34 @@ std::string guild_member::get_mention() const {
 
 guild_member& guild_member::set_nickname(const std::string& nick) {
 	this->nickname = nick;
+	this->flags |= gm_nickname_action;
+	return *this;
+}
+
+guild_member& guild_member::add_role(dpp::snowflake role_id) {
+	roles.emplace_back(role_id);
+	flags |= gm_roles_action;
+	return *this;
+}
+
+guild_member& guild_member::remove_role(dpp::snowflake role_id) {
+	roles.erase(std::remove(roles.begin(), roles.end(), role_id), roles.end());
+	flags |= gm_roles_action;
+	return *this;
+}
+
+std::string guild_member::get_nickname() {
```

Comment:
Shouldn't this return a const reference? also, shouldn't the method be const-qualified?